src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/TestClasses.cs

```diff
@@ -158,4 +158,56 @@ public struct PersonStruct
         public string FirstName { get; set; }
         public string LastName { get; set; }
     }
+
+    public class CovariantBase
+    {
+        [JsonIgnore]
+        public virtual object Id { get; }
+    }
+
+    public class CovariantDerived : CovariantBase
+    {
+        [JsonIgnore]
+        public override string Id { get; }
+    }
+
+    public class CovariantDerivedGeneric<T> : CovariantBase
+        where T : class
+    {
+        [JsonIgnore]
+        public override T Id { get; }
+    }
+
+    public class NotIgnoredPropertyBase
+    {
+        public string Id { get; set; }
+    }
+
+    public class NotIgnoredPropertyBase_NotIgnoredPropertyDerived : NotIgnoredPropertyBase
+    {
+        public new string Id { get; set; }
+    }
+
+    public class NotIgnoredPropertyBase_IgnoredPropertyDerived : NotIgnoredPropertyBase
+    {
+        [JsonIgnore]
+        public new string Id { get; set; }
+    }
+
+    public class IgnoredPropertyBase
+    {
+        [JsonIgnore]
+        public string Id { get; set; }
+    }
+
+    public class IgnoredPropertyBase_NotIgnoredPropertyDerived : IgnoredPropertyBase
+    {
+        public new string Id { get; set; }
```

Comment:
Perhaps include another test class testing overridden properties as well? I know you added one testing the generic covariant case, but I would also like a test for the simple case too.

Reply:
Done.